### DISCUSSION TITLE
Creating/adding users with enabled admin page, but without SMTP

### DISCUSSION THREAD

**Discussion by schnittchen** [dani-garcia/vaultwarden#4531] [Q&A] [answered]
Note: I could not find this question answered here, but finally figured this out myself, so I'm leaving this here for others to find.

To create a user and start using it when sending invitation emails is not possible,

* Enable the admin page, if not already done https://github.com/dani-garcia/vaultwarden/wiki/Enabling-admin-page
* On the admin page, navigate to "Users" and create a user there. It will be shown as "Invited"
* On the login page, click "Create account"
* Create an account using the exact same email. NOTE this works even when registration is disabled https://github.com/dani-garcia/vaultwarden/wiki/Disable-registration-of-new-users
* On the admin page, user is no longer shown as "Invited".

Noteworthy, for security:

* for the admin page to be securely accessed, configure HTTPS
* if you are worried about the admin token being stolen, check here https://github.com/dani-garcia/vaultwarden/wiki/Enabling-admin-page#secure-the-admin_token

**Comment by schnittchen** [comment, marked as answer]
As mentioned, answere is in the first post.

**Comment by wfjsw** [comment]
Hmm I can no longer find any Create Account button now in web vault. Probably something has changed?

![image](https://github.com/user-attachments/assets/c9e93f27-1031-4eb0-b0dc-97de979dafb2)

But using `/register` still seems to work.

**Reply by BlackDex** [reply]
Then you have registry/signups disabled in your Vaultwarden config.

**Reply by wfjsw** [reply]
Isn't it still supposed to show even if signup is disabled? 

https://github.com/dani-garcia/vaultwarden/wiki/Disable-registration-of-new-users

> Note that when SIGNUPS_ALLOWED=false, the Create Account button will still be shown in the web vault UI, but actually attempting to create an account will result in an error message. Upstream Bitwarden isn't designed to allow disabling signups, so this can't be worked around easily.

Actually I am very confused as lots of elements in the web vault seems to not in the documented place, for example the `Create Organization`. (I found it later)

**Reply by BlackDex** [reply]
Not sure which documentation you are referring to, but if it is the Vaultwarden WIki, then i think some are outdated because of the web-vault changes Bitwarden made.

Feel free to update them where needed.

Regarding the signups, that is also outdated, as we have made it able to hide elements when some features are enabled or disabled on the server side. We can now hide those elements via our custom CSS.

**Reply by BlackDex** [reply]
fyi: https://github.com/dani-garcia/vaultwarden/wiki/Customize-Vaultwarden-CSS

**Reply by TheSockCollector** [reply]
I'm running into the exact same problem, I'm hosting mine behind a university firewall, so I have essentially no chance to get a mail server running, and through multiple fresh installs with different settings I can't get a Sign Up button to show up. I don't know what /register is referring to above either, as it doesn't work on the cli in the container, and there doesn't seem to be a cli within the web interface.

**Reply by wfjsw** [reply]
`https://vault.example.com/#/register`

**Reply by BlackDex** [reply]
@TheSockCollector , of you do not see the signup link, then you probably have that feature disabled via settings.

**Reply by wfjsw** [reply]
I've edited the wiki to reflect the latest change.

**Reply by r3flow** [reply]
I tried it today with the latest version. SIGNUPS_ALLOWED=false. There is no create account link on the homepage in this case. I invited an email address in the admin interface, it shows "invited","verified". Then I open the link https://myvault.example.com/#/register, enter the invited email address and name, and I get an error message: Registration not allowed or user already exists

**Comment by sysadminpower2019** [comment]
i'm back. I originally asked this question some years ago and I just updated. I was able to create an account using the register page and login. None the less it asks me to verify via email. How do i fix that? I don't see anything in the admin page about approving it

Verify your account's email address to unlock access to all features. Send email

**Comment by Schwabsky** [comment]
Don't use their crap. If their idea is that the default option is to "invite via email" and not simply add a damn user, they have no idea about usability anyway. Absolute joke. Moreover their "fix" doesn't work. Do that and vaultwarden stays on its loading screen forever. Absolute unusable trash. 

I especially like the "invite sent successfully" when there is no mail server installed. Ridiculous.

**Comment by nox-a** [comment]
The solution only works partitially for me (I use an organization where I want to add the user to): 
- I can create a user with an invalid mail in the admin panel, which is then shown as "invited"
- After I enable registration for new users (Otherwise, as mentioned above, registration doesn't work), I can also create the account and login with it
- I however cannot add the account to the org, because I think on inviting them an email is sent to the non-existing mail address where the user has to confirm.

**Reply by stefan0xC** [reply]
Do you have mail enabled or not? If you don't have it enabled you can just add them. After registration or invitation of an existing user, you will have to confirm them as the invitation should be automatically accepted.

**Reply by nox-a** [reply]
Hello @stefan0xC 
Thanks for your quick response! Where can I check if mail is enabled?
This is how my admin panel looks like (the non-censored user is the one I am talking about)
<img width="1285" height="798" alt="admin-panel-users" src="https://github.com/user-attachments/assets/e7219026-dab2-4579-9acc-58bf42c3b977" />
My organization has the following policies:
<img width="816" height="876" alt="org-policies" src="https://github.com/user-attachments/assets/5005ff04-188f-4664-95e2-bcd1baba2eff" />

**Reply by stefan0xC** [reply]
If account recovery is available you do have mail enabled. It should be possible to temporarily deactivate mail and reinvite the user to the Organization but I haven't tested that. Also the user would have to setup a second factor because the mail as second factor would not work for them.

**Reply by nox-a** [reply]
Thanks for your hints!

So basically I have to disable Acc recovery & email (how to disable Email btw?) first.

Phew, that sounds very tedious just to add a custom user without valid Email...
Is there a chance that this could be made easier? For example, inside the Admin Console, one can create Users & add them to an org directly, maybe also set their masterpasswort already? So they could just log in without further registration.

I was able to setup a 2nd factor for the user (TOTP) after loging in, though.

**Reply by stefan0xC** [reply]
Not really. It would be easier if you would use a working mail if you have mail enabled. I mean there are also privacy concerns and security considerations regarding how this works that matter here e.g. by notifying users about a password reset by an admin, etc.

And no, I don't think you would have to disable those policies for this to work. Account recovery may not work and without a working second factor setup the user might have their membership automatically revoked but login, etc. should work with TOTP as well.

**Reply by nox-a** [reply]
Yes, I agree with you with working mail it's much easier ;) However, this is intended for users that are not necessary bound to a specific person/mail, for example for a git pipeline, build server, production user, ... where recovery is not important (In case, just delete/recreate the account).
This users could have very limited access to the DB (only keys/passwords they specifically need for their purpose).

Another (ugly) solution might be to use a temporary mail just to setup the users initially, setup TOTP and then forget about it.

**Reply by BlackDex** [reply]
The whole point is that Bitwarden is linked to email as account, which we can't (or want even) change.
Some features are just to important to let them work without email, now an admin of the host can of course bypass this, but that is something you always need to take into account with stuff like this.

Also, setting a password via the Vaultwarden Admin is not possible, as all the encryption/decryption logic is within the Bitwarden Clients, which is what the web-vault is, and that handles all those items.

Adding a user to an org is also not possible via the Vaultwarden Admin, as again, the encryption logic is within the web-vault, and that ensures the user gets the correct keys to access the org.

So, in short, you should be able to create a user with an invalid mail-address, but only if mail is disabled.
If you have mail enabled you could always try to use a `+` alias like `myname+api@domain.tld`.
1. Create the user via the Vaultwarden Admin
2. Create and Login into the account via the web-vault
3. Enable 2FA which you would like to use, but just do not use mail then
4. Invite that user into the org, and provide the correct rights to the collections

**Reply by nox-a** [reply]
Thanks for the insight & your detailed reply!

I thought it will not be so easy, I don't think it's really feasable for us to bypass the rule (because it is just very tedious for each user).
Thanks for the hint with the mail alias, I think that might be a proper solution in our case. (Or just use the temp mail to setup initially).
